flytekit/clis/sdk_in_container/run.py

```diff
@@ -0,0 +1,222 @@
+    if len(split_input) != 2:
+        raise FlyteValidationException(f"Input {file_and_workflow} must be in format '<file.py>:<worfklow>'")
+
+    filename, workflow_name = split_input
+    module = os.path.splitext(filename)[0].replace(os.path.sep, ".")
+
+    # Load code naively, i.e. without taking into account the fully qualified package name
+    wf_entity = _load_naive_entity(module, workflow_name)
+
+    if is_remote:
+        config_obj = PlatformConfig.auto()
+        client = friendly.SynchronousFlyteClient(config_obj)
+        inputs = _parse_workflow_inputs(
+            click_ctx,
+            wf_entity,
+            functools.partial(client.get_upload_signed_url, project=project, domain=domain),
+            is_remote=True,
+        )
+        _, version = script_mode.hash_file(filename)
+        remote = FlyteRemote(Config.auto(), default_project=project, default_domain=domain)
+        wf = remote.register_script(
+            wf_entity,
+            project=project,
+            domain=domain,
+            image_config=image_config,
+            destination_dir=destination_dir,
+            version=version,
+        )
+
+        options = Options(AuthRole(kubernetes_service_account=service_account))
+        execution = remote.execute(
+            wf, inputs=inputs, project=project, domain=domain, wait=wait_execution, options=options
+        )
+
+        console_url = remote.generate_console_url(execution)
+        click.secho(f"Go to {console_url} to see execution in the console.")
+
+        if dump_snippet:
+            _dump_flyte_remote_snippet(execution, project, domain)
+    else:
+        inputs = _parse_workflow_inputs(
+            click_ctx,
+            wf_entity,
+            is_remote=False,
+        )
+        print(wf_entity(**inputs))
+
+
+def _load_naive_entity(module_name: str, workflow_name: str) -> WorkflowBase:
+    """
+    Load the workflow of a the script file.
+    N.B.: it assumes that the file is self-contained, in other words, there are no relative imports.
+    """
+    flyte_ctx = context_manager.FlyteContextManager.current_context().with_serialization_settings(
+        SerializationSettings(None)
+    )
+    with context_manager.FlyteContextManager.with_context(flyte_ctx):
+        with module_loader.add_sys_path(os.getcwd()):
+            importlib.import_module(module_name)
+    return module_loader.load_object_from_module(f"{module_name}.{workflow_name}")
+
+
+def _parse_workflow_inputs(click_ctx, wf_entity, create_upload_location_fn: Optional[Callable] = None, is_remote=False):
+    args = {}
+    for i in range(0, len(click_ctx.args), 2):
+        argument = click_ctx.args[i][2:]
+        value = click_ctx.args[i + 1]
+
+        if argument not in wf_entity.interface.inputs:
+            raise FlyteValidationException(f"argument '{argument}' is not listed as a parameter of the workflow")
+
+        python_type = TypeEngine.guess_python_type(wf_entity.interface.inputs[argument].type)
```

Comment:
you can simply use - https://github.com/flyteorg/flytekit/blob/master/flytekit/core/workflow.py#L202 for getting the types and the names